### PR TITLE
New package: ppy.osu version 2024.412.1

### PR DESCRIPTION
**Pull request by Dragon1573**
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #152768

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/152924)

### CODE REVIEW

**Comment by Dragon1573** [comment]
> [!IMPORTANT]
>
> This pull request should be also reviewed by @peppy.

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ShortDescription: rhythm is just a *click* away!
+Description: |-
+  A free-to-win rhythm game. Rhythm is just a click away!
+
+  This is the future – and final – iteration of the osu! game client which marks the beginning of an open era! Currently
+  known by and released under the release codename "lazer". As in sharper than cutting-edge.
+Tags:
+- beatmap-parser
+- game
+- hacktoberfest
```
Comment (peppy) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:25]: Not sure this one should be here, doesn't really make sense for an end user.

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- osu
+- rhythm
+ReleaseNotesUrl: https://github.com/ppy/osu/releases/tag/2024.412.1
+InstallationNotes: |-
```
Comment (peppy) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:29]: Doesn't really feel like something to be mentioned in "installation notes".

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  This is the future – and final – iteration of the osu! game client which marks the beginning of an open era! Currently
+  known by and released under the release codename "lazer". As in sharper than cutting-edge.
+Documentations:
+  - DocumentLabel: Wiki
+    DocumentUrl: https://github.com/ppy/osu/wiki
```
Comment (peppy) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:34]: Might be better to link directly to https://osu.ppy.sh/wiki/en/Help_centre/Upgrading_to_lazer

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+PurchaseUrl: https://osu.ppy.sh/home/support
```
Comment (peppy) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:35]: Not sure this makes sense. The game doesn't need to be purchased. Donations are completely optional and I'd rather not have this listed on a package.

**Review by Susko3 — CHANGES_REQUESTED** [review]
As much entries as possible should be moved to the root manifest since things are unlikely to change version-by-version.

This is in hopes of avoiding duplicate manifest entries in future versions.

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml (hunk 1 above)
Comment (Dragon1573) [reply manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:25]: These tags are automatically inspected by _Komac_ from the repository homepage.

image

I would preffer attaching them here. Users can search this package by tags via `winget search`. More accurate informations in the manifest, more users can find it.

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml (hunk 3 above)
Comment (Dragon1573) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:34]:
```suggestion
  - DocumentLabel: Upgrading to lazer
    DocumentUrl: https://osu.ppy.sh/wiki/en/Help_centre/Upgrading_to_lazer
```

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml (hunk 4 above)
Comment (Dragon1573) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:35]:
```suggestion
```

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+Tags:
+- beatmap-parser
+- game
+- hacktoberfest
+- osu
+- rhythm
+ReleaseNotesUrl: https://github.com/ppy/osu/releases/tag/2024.412.1
+InstallationNotes: |-
+  This is the future – and final – iteration of the osu! game client which marks the beginning of an open era! Currently
+  known by and released under the release codename "lazer". As in sharper than cutting-edge.
```
Comment (Dragon1573) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:31]:
```suggestion
```

**Comment by Dragon1573** [comment]
> As much entries as possible should be moved to the root manifest since things are unlikely to change version-by-version.
> 
> This is in hopes of avoiding duplicate manifest entries in future versions.

Currently, manifests are version-associated. While using _Winget Releaser v2_ or other interactive tools for updating existing manifests, a full copy of outdated manifest will be generated. Contributors only needs to update a few parameters (mostly `PackageVersion`, `InstallerUrl`, `InstallerSha256` and `ReleaseNotesUrl`).

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml (hunk 1 above)
Comment (peppy) [reply manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:25]: I mean specifically `hacktoberfest`.

I've updated the repository tags, so please re-fetch them.

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+ShortDescription: rhythm is just a *click* away!
+Description: |-
+  A free-to-win rhythm game. Rhythm is just a click away!
+
+  This is the future – and final – iteration of the osu! game client which marks the beginning of an open era! Currently
+  known by and released under the release codename "lazer". As in sharper than cutting-edge.
+Tags:
+- beatmap-parser
+- game
+- hacktoberfest
```
Comment (Dragon1573) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.locale.en-US.yaml:25]:
```suggestion
```

Sync tags from the GitHub repo, as @peppy has changed them.

**Comment by Dragon1573** [comment]
@Susko3 @peppy - Is there any incorrect properties I should fixup? Could you please give approvals to my pull requests (currently at c0bb17347382ef376f08c7b56f1f1ed0bb0674c8)? Thank you ❤️

**Review by peppy — APPROVED** [review]
Seems good as far as I can tell

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.installer.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+Installers:
+- Architecture: x64
+  InstallerUrl: https://github.com/ppy/osu/releases/download/2024.412.1/install.exe
+  InstallerSha256: EEF66D09A4C2B78539A00B1C690027E2020DFFA7B4947F094BEC7231DBB8127B
+Protocols:
+- application/x-osu-beatmap
+- application/x-osu-beatmap-archive
+- application/x-osu-skin-archive
+- application/x-osu-storyboard
+- application/x-osu-replay
```
Comment (Susko3) [inline manifests/p/ppy/osu/2024.412.1/ppy.osu.installer.yaml:24]: I think protocols refers to eg. `https://`. These are media types, so they should be removed.

osu! currently supports `osu://` and `osump://`. Example: `osu://s/1`.

Here's the list of file types and protocols (URIs): https://github.com/ppy/osu/blob/a780abb0b4b14d539f1e70546cf83798fd167313/osu.Desktop/Windows/WindowsAssociationManager.cs#L41-L53

**Review by Susko3 — CHANGES_REQUESTED** [review]
Looks fine otherwise.

File: manifests/p/ppy/osu/2024.412.1/ppy.osu.installer.yaml (hunk 7 above)
Comment (Dragon1573) [reply manifests/p/ppy/osu/2024.412.1/ppy.osu.installer.yaml:24]:
```suggestion
Protocols:
- osu
- osump
FileExtensions:
- osz
- olz
- osr
- osk
```

**Comment by stephengillie** [comment]
Verify: 
image

**Review by stephengillie — APPROVED** [review]

**Comment by Dragon1573** [comment]
The package is now available again to everyone. @peppy @Susko3 🎉

```text
14:57:24 D:\...\winget-pkgs  [master ≡] 2ms pwsh> winget search -e 'ppy.osu'
名称 ID      版本       源
-------------------------------
osu! ppy.osu 2024.412.1 winget

14:59:52 D:\...\winget-pkgs  [master ≡] 3.635s pwsh> winget show -e 'ppy.osu'
已找到 osu! [ppy.osu]
版本: 2024.412.1
发布者: ppy Pty Ltd
发布服务器 URL: https://github.com/ppy
发布服务器支持 URL: https://github.com/ppy/osu/issues
描述:
  A free-to-win rhythm game. Rhythm is just a click away!

  This is the future – and final – iteration of the osu! game client which marks the beginning of an open era! Currently
  known by and released under the release codename "lazer". As in sharper than cutting-edge.
主页: https://github.com/ppy/osu
许可证: MIT
许可证 URL: https://github.com/ppy/osu/raw/master/LICENCE
版权所有: Copyright (c) 2024 ppy Pty Ltd
版权 URL: https://github.com/ppy/osu/raw/master/LICENCE
发行说明 URL: https://github.com/ppy/osu/releases/tag/2024.412.1
文档:
  Upgrading to lazer: https://osu.ppy.sh/wiki/en/Help_centre/Upgrading_to_lazer
标记：
  beatmap-parser
  game
  osu
  rhythm
安装：
  安装程序类型： exe
  安装程序 URL： https://github.com/ppy/osu/releases/download/2024.412.1/install.exe
  安装程序 SHA256： eef66d09a4c2b78539a00b1c690027e2020dffa7b4947f094bec7231dbb8127b
  发布日期: 2024-04-13
```

Manifest of version `2024.412.1` now be the "start point" of [_Winget Releaser v2_](https://github.com/marketplace/actions/winget-releaser). Members of _ppy_ can now configure a GitHub Workflow for automatically version releasing.

**Comment by peppy** [comment]
Thanks for the adjustments.

**Pull request merged**